### DISCUSSION TITLE
is there a sample of EmployeeScheduling that uses the ShiftAssignment model?

### DISCUSSION THREAD

**Discussion by mwotton** [TimefoldAI/timefold-solver#266] [Ideas]
Reading through the design patterns at https://timefold.ai/docs/timefold-solver/latest/design-patterns/design-patterns.html I noticed that using Shift or Employee as a PlanningEntity is not advised

![image](https://github.com/TimefoldAI/timefold-solver/assets/68482/8952fe2b-d7ef-4f32-9284-20180cc03049)

The quickstart still seems to have Shift as a PlanningEntity, though - is there an updated version around, or are the docs just ahead of the implementation?

**Comment by ge0ffrey** [comment]
Good find! It's confusing.

It's a paradox.
 
What is a Shift?
A) a period of time during which one or more employees need to work
B) a period of time during which one employee needs to work, with other shifts happening in the same period of time

That docs section presumes A). The quickstart presumes B). Both are fine. For clarity,  we need to rework that docs section to align with the quickstart. Please create an issue.
